crates/bevy_mikktspace/src/geometry.rs

```diff
@@ -0,0 +1,45 @@
+/// Interface for the mikktspace algorithm to query information about your geometry.
+pub trait Geometry {
+    /// Returns the number of faces.
+    fn num_faces(&self) -> usize;
+
+    /// Returns the number of vertices of a face.
+    fn num_vertices_of_face(&self, face: usize) -> usize;
+
+    /// Returns the position of a vertex.
+    fn position(&self, face: usize, vert: usize) -> [f32; 3];
+
+    /// Returns the normal of a vertex.
+    fn normal(&self, face: usize, vert: usize) -> [f32; 3];
+
+    /// Returns the texture coordinate of a vertex.
+    fn tex_coord(&self, face: usize, vert: usize) -> [f32; 2];
+
+    /// Sets the generated tangent for a vertex.
+    /// Leave this function unimplemented if you are implementing
+    /// `set_tangent_encoded`.
+    #[allow(clippy::too_many_arguments)]
+    fn set_tangent(
+        &mut self,
+        tangent: [f32; 3],
+        _bi_tangent: [f32; 3],
```

Comment:
Would it be better (or more "bevy native" or w/e) to use glam Vec2/Vec3/Vec4s instead of [f32; 2/3/4] for the stuff in this trait? I'm not sure if it really matters, but the crate already relies on glam so it's not like that'd increase potential dependencies for non-bevy library users (who also aren't using glam).